resource/environment/integration.js

```diff
@@ -7,15 +7,23 @@ var environment = require( "../../src/data/nedb/environment" );
 var statusIntervals = {};
 var pendingUpgrade = {};
 
+function onFinish( slack, channels, env, service ) {
+	var message = format( "The upgrade of the service %s in environment %s has been finalized.", service.name, env.name );
+	_.each( channels, function( channel ) {
+		slack.send( channel, message );
+	} );
+}
+
 function onServiceList( env, channels, slack, services ) {
 	_.each( services, function( service ) {
 			if ( pendingUpgrade[ service.id ] && service.state === "upgraded" ) {
-				var message = format( "The service %s in environment %s has upgraded successfully, amigo.",
+				var message = format( "The service %s in environment %s has upgraded successfully, amigo. Finalizing upgrade now.",
 					service.name, env.name );
 				_.each( channels, function( channel ) {
 					slack.send( channel, message );
 				} );
 				delete pendingUpgrade[ service.id ];
+				service.finish().then( onFinish.bind( null, slack, channels, env ), console.error );
```

Comment:
probably want to have a handler that returns a status code like the other onError methods do